src/libstd/io/error.rs

```diff
@@ -548,15 +548,6 @@ impl error::Error for Error {
         }
     }
 
-    #[allow(deprecated)]
-    fn cause(&self) -> Option<&dyn error::Error> {
-        match self.repr {
-            Repr::Os(..) => None,
-            Repr::Simple(..) => None,
-            Repr::Custom(ref c) => c.error.cause(),
-        }
```

Comment:
This risks being a noticeable change since we are calling out to a user-supplied .cause() vs .source() here.

Reply:
Good catch! I almost expected there was some aspect of this I missed :sweat_smile: 
I removed the last commit.